packages/aws-cdk-lib/aws-ec2/lib/instance-types.ts

```diff
@@ -326,6 +326,11 @@ export enum InstanceClass {
    */
   R7GD = 'r7gd',
 
+  /**
+   * Memory optimized instances based on Intel Xeon Scalable (Sapphire Rapids) processors, 7th generation
```

Comment:
Lets change this to not be the same as r7iz. Or if the better change would be to that doc string, do that.

Reply:
i agree, i had difficulty finding the nuanced difference between the `r7i` and `r7iz`, unfortunately you're correct, the `MEMORY7_INTEL` should have gone to the `r7i` instead of the `r7iz`, but I had thought the `r7iz` was the replacement for the `r7i` since it used the same processor generation as the `c7i` and `m7i`. 

I'm totally open to suggestions, but happy to take a stab at it for community review.

Reply:
Yes, that mistake is on me. Do what you think looks good and we can go from there!